test/uk/gov/hmrc/channelpreferences/controllers/PreferenceControllerSpec.scala

```diff
@@ -328,6 +334,42 @@ class PreferenceControllerSpec extends PlaySpec with ScalaCheckPropertyChecks wi
 
   }
 
+  "Calling update contact" should {
+    "return OK when the call to EIS succeed" in new TestSetup {
+      when(mockEISConnector.updateContactPreference(anyString(), anyBoolean(), anyString()))
+        .thenReturn(Future.successful(Ok))
+
+      val fakeRequest = FakeRequest("POST", "/path")
+      val result = controller.update("itsa", "true").apply(fakeRequest)
+      status(result) mustBe Status.OK
+    }
+
+    "return INTERNAL_SERVER_ERROR (500) when the call to EIS fails" in new TestSetup {
+      when(mockEISConnector.updateContactPreference(anyString(), anyBoolean(), anyString()))
+        .thenReturn(Future.successful(InternalServerError))
+
+      val fakeRequest = FakeRequest("POST", "/path")
+      val result = controller.update("itsa", "true").apply(fakeRequest)
+      status(result) mustBe Status.INTERNAL_SERVER_ERROR
+    }
+
+    "return BAD_REQUEST when the key is not supported" in new TestSetup {
+
+      val fakeRequest = FakeRequest("POST", "/path")
+      private val unsupportedKey = "unsupportedKey"
+      val result = controller.update(unsupportedKey, "true").apply(fakeRequest)
+      status(result) mustBe Status.BAD_REQUEST
+    }
+
+    "return BAD_REQUEST when the status is not supported" in new TestSetup {
+
+      val fakeRequest = FakeRequest("POST", "/path")
+      private val unsupportedStatus = "unsupportedStatus"
+      val result = controller.update("itsa", unsupportedStatus).apply(fakeRequest)
+      status(result) mustBe Status.BAD_REQUEST
```

Comment:
here too, we could assert that the message returned is indeed "Unexpected status for itsa unsupportedStatus"

Reply:
added in https://github.com/hmrc/channel-preferences/pull/34/commits/48044e0ccc91bc2192a9f8b9f6a50d26afc4ae96